src/pages/Encounters/ReportBuilder/HeaderBuilder.tsx

```diff
@@ -0,0 +1,865 @@
+  activeElement: number;
+  setActiveElement: (elementIndex: number) => void;
+}
+
+const RowButtons = ({
+  columnIndex: rowIndex,
+  size = "sm",
+  handleAddElement,
+}: {
+  columnIndex: number;
+  size?: "sm" | "xs" | "default" | "lg" | "icon";
+  handleAddElement: (rowIndex: number, type: string) => void;
+}) => {
+  const { t } = useTranslation();
+  return (
+    <div className="flex flex-col sm:flex-row items-center rounded-lg border overflow-clip">
+      <Button
+        type="button"
+        size={size}
+        variant="ghost"
+        onClick={() => handleAddElement(rowIndex, "text")}
+        className="w-full flex flex-col items-center gap-1 px-5 py-8 rounded-none border-r border-gray-200 bg-green-50 text-green-900 hover:bg-green-200/80 hover:text-green-900"
+      >
+        <CareIcon icon="l-text" className="w-4 h-4" />
+        {t("REPORT_BUILDER_ADD_TEXT")}
+      </Button>
+      <Button
+        type="button"
+        size={size}
+        variant="ghost"
+        onClick={() => handleAddElement(rowIndex, "image")}
+        className="w-full flex flex-col items-center gap-1 px-5 py-8 rounded-none border-t sm:border-t-0 border-r border-gray-200 bg-green-50 text-green-900 hover:bg-green-200/80 hover:text-green-900"
+      >
+        <CareIcon icon="l-image-v" className="w-4 h-4" />
+        {t("REPORT_BUILDER_ADD_IMAGE")}
+      </Button>
+      <Button
+        type="button"
+        size={size}
+        variant="ghost"
+        onClick={() => handleAddElement(rowIndex, "rule")}
+        className="w-full flex flex-col items-center gap-1 px-5 py-8 rounded-none border-t sm:border-t-0 border-r border-gray-200 bg-green-50 text-green-900 hover:bg-green-200/80 hover:text-green-900"
+      >
+        <CareIcon icon="l-minus" className="w-4 h-4" />
+        {t("REPORT_BUILDER_ADD_RULE")}
+      </Button>
+      <Button
+        type="button"
+        size={size}
+        variant="ghost"
+        onClick={() => handleAddElement(rowIndex, "datetime")}
+        className="w-full flex flex-col items-center gap-1 px-4 py-8 rounded-none border-t sm:border-t-0 border-r border-gray-200 bg-green-50 text-green-900 hover:bg-green-200/80 hover:text-green-900"
+      >
+        <CareIcon icon="l-calender" className="w-4 h-4" />
+        {t("REPORT_BUILDER_ADD_DATETIME")}
+      </Button>
+    </div>
+  );
+};
+
+const HeaderRow = React.memo(function HeaderRow({
+  columnIndex: columnIndex,
+  column: column,
+  form,
+  onRemoveRow,
+  onAddElement,
+  onRemoveElement,
+  activeElement,
+  setActiveElement,
+}: HeaderRowProps) {
+  const { t } = useTranslation();
+
+  const toggleElement = useCallback(
+    (index: number) => {
+      setActiveElement(index);
+    },
+    [setActiveElement],
+  );
+
+  const handleAddElement = useCallback(
+    (rowIndex: number, type: string) => {
+      onAddElement(rowIndex, type);
+    },
+    [onAddElement],
+  );
+
+  const getElementIcon = (type: string) => {
```

Comment:
nitpick: it's good to have strictest types.

```suggestion
  const getElementIcon = (type: HeaderElementType["type"]) => {
```